src/controllers/rpc/publish-rpc-controller.js

```diff
@@ -18,30 +18,35 @@ class PublishController extends BaseController {
         const command = { sequence: [], delay: 0, transactional: false, data: {} };
         let dataSource;
         const [handleInitCommand, handleRequestCommand] = this.getCommandSequence(protocol);
-        switch (messageType) {
-            case NETWORK_MESSAGE_TYPES.REQUESTS.PROTOCOL_INIT:
-                dataSource = message.data;
-                command.name = handleInitCommand;
-                command.period = 5000;
-                command.retries = 3;
+        try {
+            switch (messageType) {
+                case NETWORK_MESSAGE_TYPES.REQUESTS.PROTOCOL_INIT:
+                    dataSource = message.data;
+                    command.name = handleInitCommand;
+                    command.period = 5000;
+                    command.retries = 3;
 
-                break;
-            case NETWORK_MESSAGE_TYPES.REQUESTS.PROTOCOL_REQUEST:
-                // eslint-disable-next-line no-case-declarations
-                dataSource = await this.operationIdService.getCachedOperationIdData(operationId);
-                await this.operationIdService.cacheOperationIdData(operationId, {
-                    assertionId: dataSource.assertionId,
-                    assertion: message.data.assertion,
-                });
-                command.name = handleRequestCommand;
-                command.data.keyword = message.data.keyword;
-                command.data.agreementId = dataSource.agreementId;
-                command.data.agreementData = dataSource.agreementData;
-                break;
-            default:
-                throw Error('unknown message type');
+                    break;
+                case NETWORK_MESSAGE_TYPES.REQUESTS.PROTOCOL_REQUEST:
+                    // eslint-disable-next-line no-case-declarations
+                    dataSource = await this.operationIdService.getCachedOperationIdData(
+                        operationId,
+                    );
+                    await this.operationIdService.cacheOperationIdData(operationId, {
+                        assertionId: dataSource.assertionId,
+                        assertion: message.data.assertion,
+                    });
+                    command.name = handleRequestCommand;
+                    command.data.keyword = message.data.keyword;
+                    command.data.agreementId = dataSource.agreementId;
+                    command.data.agreementData = dataSource.agreementData;
+                    break;
+                default:
+                    throw Error('unknown message type');
+            }
+        } catch (error) {
+            this.logger.error(error.message);
```

Comment:
When logging an error add more context, where error happened, don't just log error.message.
When publish or any other operation is starting we are emitting start events, if for any reason operation cannot be completed we need to publish end event with error message. Update your code to better handle error in operation

Reply:
updated it, please take a look again